examples/gadgets/operators/otel_metrics/main.go

```diff
@@ -0,0 +1,168 @@
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	apihelpers "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api-helpers"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	otelmetrics "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/otel-metrics"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/simple"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local"
+)
+
+func do() error {
+	var producer func()
+
+	// We will simulate generating metrics for three nodes; these strings will serve as the only key for the metrics
+	nodes := []string{"node1", "node2", "node3"}
+
+	metricsGenerator := simple.New("myHandler",
+		simple.OnInit(func(gadgetCtx operators.GadgetContext) error {
+			ds, err := gadgetCtx.RegisterDataSource(datasource.TypeSingle, "metrics")
+			if err != nil {
+				return err
+			}
+			ds.AddAnnotation(otelmetrics.AnnotationMetricsExport, "true")
+
+			// The node name will be used as key
+			node, err := ds.AddField(
+				"node",
+				api.Kind_String,
+				datasource.WithAnnotations(map[string]string{
+					otelmetrics.AnnotationMetricsType:        otelmetrics.MetricTypeKey,
+					otelmetrics.AnnotationMetricsDescription: "Node name",
+				}),
+			)
+
+			// Latency will be recorded as histogram
+			latency, err := ds.AddField(
+				"latency",
+				api.Kind_Uint64,
+				datasource.WithAnnotations(map[string]string{
+					otelmetrics.AnnotationMetricsType:        otelmetrics.MetricTypeHistogram,
+					otelmetrics.AnnotationMetricsDescription: "Latency",
+				}),
+			)
+
+			// Memory will be recorded as gauge
+			mem, err := ds.AddField(
+				"memory",
+				api.Kind_Uint64,
+				datasource.WithAnnotations(map[string]string{
+					otelmetrics.AnnotationMetricsType:        otelmetrics.MetricTypeGauge,
+					otelmetrics.AnnotationMetricsDescription: "Memory usage",
+				}),
+			)
+
+			// Every second the counter gets increased by 1
+			ctr, err := ds.AddField(
+				"ctr",
+				api.Kind_Uint64,
+				datasource.WithAnnotations(map[string]string{
+					otelmetrics.AnnotationMetricsType:        otelmetrics.MetricTypeCounter,
+					otelmetrics.AnnotationMetricsDescription: "Number of metric events",
+				}),
+			)
+
+			producer = func() {
+				ticker := time.NewTicker(time.Second)
+				for {
+					select {
+					case <-ticker.C:
+						for _, n := range nodes {
+							// emit new metrics
+							metrics, err := ds.NewPacketSingle()
+							if err != nil {
+								gadgetCtx.Logger().Warnf("failed to create packet: %v", err)
+								continue
+							}
+
+							node.PutString(metrics, n)
+
+							latency.PutUint64(metrics, uint64(rand.Intn(2000)))
+
+							var m runtime.MemStats
+							runtime.ReadMemStats(&m)
+							mem.PutUint64(metrics, m.TotalAlloc)
+
+							ctr.PutUint64(metrics, 1)
+
+							ds.EmitAndRelease(metrics)
+						}
+					case <-gadgetCtx.Context().Done():
+						break
```

Comment:
visual studio code: "ineffective break statement. Did you mean to break out of the outer loop? (SA4011)go-staticcheck"